### ISSUE TITLE
Map.values() not really an iterable sequence?

### ISSUE THREAD

**Issue by leidegre** [facebook/immutable-js#791]
Basically, this works `map.map((v) => v).flatten().toArray()` but this does not `map.values().flatten().toArray()` because `flatten` is not defined on whatever iterable is returned by `values()` but according to the docs I cannot see why that is? Shouldn't `map.map((v) => v)` be equivalent to `map.values()`?

You can test run it in this JSFiddle https://jsfiddle.net/372c9rxk/1/ that I prepared.

**Comment by pandanoir** [comment]
map.value returns Iterator, not Iterable.

Iterator doesn't have flatten method.

**Comment by leidegre** [comment]
@pandanoir Thanks for clearing that up.

Why this difference though, what's the point between having a `Iterator` and an `Iterable`, or more importantly why are there APIs similar but not identical? Is there are good reason for this?

I have found myself slipping here more than once and I really can't fathom why `map.map((v)=>v)` wouldn't be the equivalent of `map.values()`.

Oh, this is intresting, https://github.com/facebook/immutable-js/blob/95335c36f2c4a3635b2ee139d906647c1d997279/dist/immutable.d.ts#L2528 If it is not part of the library I don't think it should be documented or there should be some caverat about that the Iterator is not compatible with other Immutable JS operations.

**Comment by pandanoir** [comment]
Iterable is Iterator. In Iterator pattern, Iterator only has next() method while the object which implements Iterator has many unique and useful methods.

Iterable is Iterator, so Iterator doesn't have `flatten()` while Iterable has.

C++ Iterator may help you understand Iterator.

**Comment by leidegre** [comment]
@pandanoir No, the whole confusion stems from a name change that occurred between v2 and v3. Where `Sequence` was changed to `Iterable`, which is almost identical to `Iterator`. The `keys`, `values` and `entries` are actually ES6 APIs that follow ES6 iterator protocols. **This was a surprise to me.** I've added a clarification about that in the documentation as I think anyone could easily make this mistake.
